src/precice/impl/ValidationMacros.hpp

```diff
@@ -90,37 +89,31 @@
  *
  * @attention Do not use this macro directly!
  */
-#define PRECICE_VALIDATE_DATA_ID_IMPL(id)                                                                                                           \
-  PRECICE_CHECK(std::any_of(_dataIDs.begin(), _dataIDs.end(), [id](const typename decltype(_dataIDs)::value_type &meshkv) {                         \
-                  return std::any_of(meshkv.second.begin(), meshkv.second.end(), [id](const typename decltype(meshkv.second)::value_type &datakv) { \
-                    return datakv.second == id;                                                                                                     \
-                  });                                                                                                                               \
-                }),                                                                                                                                 \
+#define PRECICE_VALIDATE_DATA_ID_IMPL(id) \
+  PRECICE_CHECK(_accessor->hasData(id),   \
                 "The given Data ID \"{}\" is unknown to preCICE.", id);
 
 /** Implementation of PRECICE_REQUIRE_DATA_READ()
  *
  * @attention Do not use this macro directly!
  */
-#define PRECICE_REQUIRE_DATA_READ_IMPL(id)                                                                                  \
-  PRECICE_VALIDATE_DATA_ID_IMPL(id)                                                                                         \
-  DataContext &context = _accessor->dataContext(id);                                                                        \
-  PRECICE_CHECK((_accessor->isDataUsed(id) && _accessor->isDataRead(id)),                                                   \
-                "This participant does not use Data \"{}\", but attempted to read it. "                                     \
-                "Please extend the configuarion of partiticpant \"{}\" by defining <read-data mesh=\"{}\" name=\"{}\" />.", \
-                context.getName(), _accessorName, context.mesh->getName(), context.getName());
+#define PRECICE_REQUIRE_DATA_READ_IMPL(id)                                                                                      \
```

Comment:
May I ask you to provide a comment, maybe at the top of this file, why you split all the macros in an implementation part and a definition?